clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -909,6 +909,39 @@ public void ensureValid(String name, Object o) {
         }
     }
 
+    public static class ValidName implements Validator {
+        final char[] illegalCharacters = {'/', ' ', '\\', '\n'};
+
+        public static ValidName validName() {
+            return new ValidName();
+        }
+
+        @Override
+        public void ensureValid(String name, Object value) {
+            String s = (String) value;
+
+            if (s == null) {
+                // This can happen during creation of the config object due to no default value being defined for the
+                // name configuration - a missing name parameter is caught when checking for mandatory parameters,
+                // thus we can ok a null value here
+                return;
+            } else if (s.isEmpty()) {
+                throw new ConfigException(name, value, "Name may not be empty");
+            }
+
+            // Check name string for illegal characters
```

Comment:
Hi Soenke, I'd also make a comment if you don't mind.
Have you considered using precompiled regexes to match against the connector names?